src/sql/parts/notebook/outputs/renderers.ts

```diff
@@ -265,6 +271,88 @@ export function renderLatex(
 	return Promise.resolve(undefined);
 }
 
+/**
+ * Render DataResource as a grid into a host node.
+ *
+ * @params options - The options for rendering.
+ *
+ * @returns A promise which resolves when rendering is complete.
+ */
+export function renderDataResource(
+	options: renderDataResource.IRenderOptions
+): Promise<void> {
+	// Unpack the options.
+	let { host, source } = options;
+	let sourceObject: IDataResource = JSON.parse(source);
+
+	// Before doing anything, avoid re-rendering the table multiple
+	// times (as can be the case when going untrusted -> trusted)
+	while (host.firstChild) {
+		host.removeChild(host.firstChild);
+	}
+
+	// now create the table
+	let tableContainer = document.createElement('div');
+	tableContainer.className = 'notebook-cellTable';
+
+	const ROW_HEIGHT = 29;
+	const BOTTOM_PADDING_AND_SCROLLBAR = 14;
+	let tableResultsData = new TableDataView();
+	let columns: string[] = sourceObject.schema.fields.map(val => val.name);
+	// Table object requires passed in columns to be of datatype Slick.Column
+	let columnsTransformed = transformColumns(columns);
+
+	// In order to show row numbers, we need to put the row number column
+	// ahead of all of the other columns, and register the plugin below
+	let rowNumberColumn = new RowNumberColumn({ numberOfRows: source.length });
+	columnsTransformed.unshift(rowNumberColumn.getColumnDefinition());
+
+	tableResultsData.push(transformData(sourceObject.data, columns));
+
+	let detailTable = new Table(tableContainer, {
+		dataProvider: tableResultsData, columns: columnsTransformed
+	}, { rowHeight: ROW_HEIGHT,
+		forceFitColumns: false,
+		defaultColumnWidth: 120
```

Comment:
Where are the defaults coming from?

Reply:
They match the query editor's values (which unfortunately are hardcoded as well). Any ideas where I can put this in a common place?

Reply:
@anthonydresser can you advise?